### PR TITLE
first attempt and ID based auth example

### LINKED ISSUE #5: [TODO]: node id based authorisation
create code example in nodejs which implements id based authorisation

### PR DESCRIPTION
**Pull request by john681611**
Fixes #5

### CODE REVIEW

File: skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md [hunk 1]
```diff
@@ -5,13 +5,81 @@
 - [Considerations](#considerations)
 
 ## General
-TBA
+Based off this post on [Medium](https://medium.com/@evangow/server-authentication-basics-express-sessions-passport-and-curl-359b7456003d);
```
Comment (northdpole) [inline skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md]: let's not reference medium posts

File: skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md [hunk 2]
```diff
@@ -5,13 +5,81 @@
...
+
+// create the server
+const app = express();
+
+// add & configure middleware
+app.use(bodyParser.urlencoded({ extended: false }))
+app.use(bodyParser.json())
+app.use(session({
+  store: new FileStore(),
+  secret: process.env.sessionKey, //always use environment variables to pass in keys.
```
Comment (northdpole) [inline skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md]: can we add: even better use a secrets store such as vault?

File: skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md [hunk 3]
```diff
@@ -5,13 +5,81 @@
...
+//Login excluded
+
+app.get('/post',  passport.authenticate('local', { failureRedirect: '/login' }), //authenticates the user using session
+   async function(req, res) {
+	  	const data = req.body;
+	  	let pool = await sql.connect(config)
+		let result = await pool.request()
+			.input('user_id', sql.Int, req.user.id) //sql.Int validates that only a integer value can be in the variable
+			.input('file_id', sql.Int, data.id)
+            .query('select * from file_access where user_id = @user_id and file_id = @file_id'); //variables inlined into sql query
```
Comment (northdpole) [inline skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md]: sqli possible?

File: skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md (hunk 2 above)
Comment (john681611) [reply skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md]: You can but it's making it extra complex, environment variables are what is used as a standard.

File: skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md (hunk 3 above)
Comment (john681611) [reply skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md]: honestly not used SQLI just used what I knew. Can convert if needed.

Comment (northdpole) [reply skf/markdown/code_examples/nodejs-express/17-code_example--Identifier_based_authorization--.md]: sorry, posted this quite late(?)
Nevermind, figured out how it works, its parameterized queries

**Pull request merged**